Pokemon/Pokemon/PokemonTableViewController.m

```diff
@@ -0,0 +1,506 @@
+                          @"Wigglytuff",
+                          @"Meowth",
+                          @"Persian",
+                          @"Farfetch",
+                          @"Doduo",
+                          @"Dodrio",
+                          @"Lickitung",
+                          @"Chansey",
+                          @"Kangaskhan",
+                          @"Tauros",
+                          @"Ditto",
+                          @"Eevee",
+                          @"Porygon",
+                          @"Snorlax",
+                          @"Geodude",
+                          @"Graveler",
+                          @"Golem",
+                          @"Onix",
+                          @"Rhyhorn",
+                          @"Rhydon",
+                          @"Aerodactyl",
+                          @"Clefairy",
+                          @"Clefable",
+                          @"Mankey",
+                          @"Primeape",
+                          @"Machop",
+                          @"Machoke",
+                          @"Machamp",
+                          @"Hitmonlee",
+                          @"Hitmonchan"];
+
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+#warning Potentially incomplete method implementation.
+    
+    NSArray *keys = [self.pokemon allKeys];
+    
+    return keys.count;
+    
+    
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+#warning Incomplete method implementation.
+    
+    NSArray *keys = [self.pokemon allKeys];
+    NSString *words = keys[section];
+    NSArray *words2 = [self.pokemon objectForKey:words];
+    
+    return words2.count;
+    
+    
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"pokemonCell" forIndexPath:indexPath];
+    
+    NSArray *keys = [self.pokemon allKeys];
+    NSString *words = keys[indexPath.section];
+    NSArray *words2 = [self.pokemon objectForKey:words];
+    
+    NSString *pokemonNames = [words2 objectAtIndex:indexPath.row];
+    
+    cell.textLabel.text = pokemonNames;
+    cell.imageView.image = [UIImage imageNamed:pokemonNames];
+    
+    return cell;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
+
+    NSArray *key = [self.pokemon allKeys];
+    NSString *words = key [section];
+    
+    return words;
+}
+
+- (UIView *) tableView:(UITableView *) tableView viewForHeaderInSection:(NSInteger)section{
+
+    if (section == 0) {
+        
+    UISegmentedControl *segmentedControl = [[UISegmentedControl alloc] initWithItems:@[@"Alphabetically", @"Type"]];
```

Comment:
Style nitpick: This line should be indented four more spaces. You can automatically re-indent in Xcode with ctrl+i.